app/controllers/sessions_controller.rb

```diff
@@ -0,0 +1,18 @@
+class SessionsController < ApplicationController
+  skip_before_action :authenticate_user!, only: [:create]
+
+  def create
+    session.update(
+      current_user: User.build_from(auth_hash),
```

Comment:
I see now that User is not actually an activerecord model... lower down in the diff. Leaving the comment for posterity.

I'm still pondering if there's security implications to this though :)